docs/src/pages/playground.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useEffect, useRef, useState } from 'react'
+import Layout from '@theme/Layout'
+import sdk from '@stackblitz/sdk'
+import styles from './playground.module.css'
+import { useColorMode } from '@docusaurus/theme-common'
+import { usePluginData } from '@docusaurus/useGlobalData'
+
+interface Project {
+	categoryTitle: string
+	subCategory: Array<CategoryItems>
+}
+
+interface CategoryItems {
+	name: string
+	templates: Array<Templates>
+}
+
+interface Templates {
+	title: string
+	desc: string
+	link: string
+	idx: number
+}
+
+const useRoboData = () => {
+	const { env } = usePluginData('@robojs/docusaurus') as { env: { port: string } }
+
+	return {
+		hostRelative: process.env.NODE_ENV === 'development' ? 'http://localhost:' + env.port : ''
+	}
+}
+
+function Playground() {
+	const [projectLink, setProjectLink] = useState('Wave-Play/robo.js/tree/main/templates/starter-app-js-react')
+	const [dropdown, setDropdown] = useState(false)
+	const [embedLoading, setEmbedLoading] = useState(true)
+	const { colorMode } = useColorMode()
+	const [fetchError, setFetchError] = useState<null | string>(null)
+	const [templates, setTemplates] = useState<Project[] | null>(null)
+	const [selectedTemplateIndex, setSelectedTemplateIndex] = useState(0)
+	const { hostRelative } = useRoboData()
+
+	const embedDiv = useRef<HTMLIFrameElement | null>(null)
+	const editor = useRef(null)
+
+	useEffect(() => {
+		const fetchTemplates = async () => {
+			const res = await fetch(hostRelative + '/api/templates')
+
+			if (res.status !== 20) {
+				setFetchError(res.statusText)
+			}
+
+			if (res.ok && res.status === 200) {
```

Comment:
Why checking for status 200 specifically? Do we expect the API result to include other 2xx?